lib.php

```diff
@@ -512,6 +514,44 @@ public function supported_returntypes() {
             return FILE_INTERNAL;
         }
     }
+    
```

Comment:
Isn't this code just copied over from the preferences_form? Is that really necessary?